### ISSUE TITLE
Multi-line (i.e. fenced) code-block start should not include the code itself

### ISSUE THREAD

**Issue by serious-angel** [Hacksore/discord-md#2]
Greetings,

I like the current design, and especially the current colors on the main page:

![image](https://github.com/user-attachments/assets/1008ab89-e300-4caa-8a6f-14b2f61bf920)

❤️

---

The content of the project, however, is not sufficient nor correct enough with alternatives I believe.

For instance, the following:

https://github.com/Hacksore/discord-md/blob/9f4c5d67047edda35a5dae057b7689a916668e14/apps/web/src/content/feature/code-blocks.md?plain=1#L10-L12

![image](https://github.com/user-attachments/assets/2f9f63cf-9128-478b-9e32-077098bc9672)

This is not a standard Markdown format, including [CommonMark](<https://commonmark.org/>), of such code-block also called [fenced code blocks](<https://www.markdownguide.org/extended-syntax/#fenced-code-blocks>) ([RFC7764](<https://datatracker.ietf.org/doc/html/rfc7764.html#section-4.5>)).  
Some would call such [escaping backticks](<https://www.markdownguide.org/basic-syntax/#escaping-backticks>) at this point.

---

In other words, it should not be:

````
```haikus...```
````
...but rather:

````
```
haikus...
```
````

Since, this also allows to set the syntax (e.g. [CoffeeScript](<https://coffeescript.org/>)):

````
```coffee
haikus...
```
````

Best and kind regards

**Comment by Hacksore** [comment]
Hey @artshade thanks for the issue!

let's use this issue to make a preview component so that we can get rid of the images.

**Comment by Hacksore** [comment]
so played around with creating an astro component that would let us have markdown previewer.

It's pretty bad atm because I am using `marked` but maybe it could be the way for now at least?

https://discord-md.vercel.app/feature/misc#testing-md-with-astro-comp

The other option is figure out how to create a custom rehype/remark plugin do have something like this in `.md` files.
<img width="383" alt="image" src="https://github.com/user-attachments/assets/0ab3466e-ff11-44ab-929e-850137d6cbc4">

and it would output something like this.

<img width="1020" alt="image" src="https://github.com/user-attachments/assets/5283b992-d1ec-4f51-ad76-89c8707cadc8">

**Comment by Hacksore** [comment]
Pushed a few updates but also found that this table on [mdxjs.com](https://mdxjs.com/table-of-components/) is almost what trying to build 😂.

The only difference instead of showing HTML on the far right it'd be  what it looks like in discord.
<img width="1596" alt="image" src="https://github.com/user-attachments/assets/c7c580e0-1edd-4ec1-b4c3-7301718f58b5">

how they do it for ref https://raw.githubusercontent.com/mdx-js/mdx/main/docs/table-of-components.mdx

**Comment by Hacksore** [comment]
I think this is ok now with the preview component, if you find another bug raise a new issue.

feel free to break stuff on the playground.
https://discord-md.vercel.app/play
